### PR TITLE
Enable 3D active item processing

### PR DESCRIPTION
**Pull request by Coolthulhu**
Enabling itself is actually only about ~5 lines of code. Rest is fixing explosions and earlier bugs.

Implemented the bailing out on smashing improper tent tile. Now instead of saying "rrrrip", it doesn't say anything... Didn't fix indestructible tents here.

Stopped explosions from being drawn on outside current z-level and stopped them from affecting things outside current z-level.

Currently sound directions are inverted on z axis. ie. sounds from above have descriptions as if they were from below. Will update it later.

Includes #12318

### CODE REVIEW

**Comment by BevapDin** [comment]
> Currently sound directions are inverted on z axis

Yeah, I was about to comment on that. The z-values in the `direction` enum are inverted.

File: src/game.cpp [hunk 1]
```diff
@@ -13210,11 +13230,11 @@ void game::shift_monsters( const int shiftx, const int shifty, const int shiftz
         if( shiftx != 0 || shifty != 0 ) {
             critter.shift( shiftx, shifty );
         }
-        // TODO: with z-levels, this can be removed, instead shift the critter
-        // along shiftz, too. Than make the 3D-inbounds check.
-        if( shiftz == 0 && m.inbounds( critter.posx(), critter.posy() ) ) {
+
+        if( m.has_zlevels() && m.inbounds( critter.pos() ) ) {
```
Comment (BevapDin) [inline src/game.cpp]: No it will always despwan if `m.has_zlevels` returns false. I think this condition really needs to include the shiftz parameter.

Comment (Coolthulhu) [inline src/game.cpp]: Like this maybe?

Comment (BevapDin) [inline src/game.cpp]: Looks fine.

A completely unrelated bug:
The scrolling combat text does not consider z-levels at all.  If a zombies is exactly above my character and attacks it, the SCT only considers x,y for the direction and ends up storing `CENTER` as movement direction.

This results in an infinite loop in `game::get_player_input`. Calling `advanceStepOffset` there is supposed to change the position of the text, but because the direction along which the text should move is `CENTER`, it does not move at all.

So in short: the SCT overlays creature, `advanceStepOffset` is called to move it away, but the position is not affects by this, so the same creature still interferes with the text.

I assume the onscreen-text should never have the direction `CENTER` or `CENTER_ABOVE` or `CENTER_BELOW`, similar `iDirX` and `iDirY` should not both be 0 (that's actually more important than the direction).

**Comment by BevapDin** [comment]
Besides the SCT thing, it looks good.

It's a pity the are no nice test areas (the evac shelter needs a second floor). Also: the different locations of the two stairs in the basement and in the ground level of the evac shelter are now really obvious and disturbing.

**Comment by Coolthulhu** [comment]
I could try to fix the STC thing by forcing a direction if it's outside allowed range. Send it now or in a separate PR?

**Comment by BevapDin** [comment]
> Send it now or in a separate PR?

Now would be good, I don't want to merge this as it is currently because that infinite loop is quite annoying.

**Comment by Coolthulhu** [comment]
Sent an attempt at fix.

If you want the zeds to stop biting, Debug Invincibility mutation prevents them from dealing damage (and thus causing STC). Also useful for debugging explosives.
